lib/algolia.ex

```diff
@@ -123,8 +123,13 @@ defmodule Algolia do
 
   defp default_middleware(opts) do
     [
-      Algolia.Middleware.Telemetry,
-      {Algolia.Middleware.Headers, Keyword.take(opts, [:api_key, :application_id])},
+      Algolia.Middleware.Response,
+      Tesla.Middleware.Telemetry,
```

Comment:
I think we can pass in `disable_legacy_event` and add `metadata: %{client: "algolia-elixir"}`